File: bank.cpp
//Write a program of bank class

#include<iostream>
#include<string.h>

using namespace std;

class bank
{
  string bank_name;
  long acc_no;
  string type_of_acc;
  float balance;
  int amount;

  public:
  void getinfo();
  void displayinfo();
  void credit();
  void debit();
};

void bank::getinfo()
{
  cout<<"Enter Bank Name : "<<endl;
  cin>>bank_name;
  cout<<"Enter Account Number : "<<endl;
  cin>>acc_no;
  cout<<"Enter Type Of Account : "<<endl;
  cin>>type_of_acc;
  cout<<"Enter Balance : "<<endl;
  cin>>balance;
  cout<<"Enter Amount : "<<endl;
  cin>>amount;
}

void bank::displayinfo()
{
  cout<<"Bank Name : "<<bank_name<<endl;
  cout<<"Account Number : "<<acc_no<<endl;
  cout<<"Type Of Account : "<<type_of_acc<<endl;
  cout<<"Balance : "<<balance<<endl;
  cout<<"Amount you want to be credited or debited : "<<amount<<endl;
}

void bank::credit()
{
  int credit;
  credit=balance+amount;
  cout<<"Total amount after credited :"<<credit<<endl;
}

void bank::debit()
{
  int debit;
  if((balance>amount)&&(balance>2000))
  {
    cout<<"You can withdraw the amount"<<endl;
    debit=balance-amount;
    cout<<"Total balance after the withdrawl :"<<debit<<endl;
  }  
  else
  {
    cout<<"You have insufficient amount in your account"<<endl;
  }
}
int main()
{
  bank b1;
  b1.getinfo();
  b1.displayinfo();
  b1.credit();
  b1.debit();
  return 0;
}

File: Bank.java
import java.util.Scanner;

class Bank {
    String bankName;
    long accNo;
    String typeOfAcc;
    float balance;
    int amount;

    public void getInfo() {
        Scanner input = new Scanner(System.in);
        System.out.println("Enter Bank Name: ");
        bankName = input.nextLine();
        System.out.println("Enter Account Number: ");
        accNo = input.nextLong();
        input.nextLine(); // consume newline character
        System.out.println("Enter Type Of Account: ");
        typeOfAcc = input.nextLine();
        System.out.println("Enter Balance: ");
        balance = input.nextFloat();
        System.out.println("Enter Amount: ");
        amount = input.nextInt();
    }

    public void displayInfo() {
        System.out.println("Bank Name: " + bankName);
        System.out.println("Account Number: " + accNo);
        System.out.println("Type Of Account: " + typeOfAcc);
        System.out.println("Balance: " + balance);
        System.out.println("Amount you want to be credited or debited: " + amount);
    }

    public void credit() {
        float credit = balance + amount;
        System.out.println("Total amount after credited: " + credit);
    }

    public void debit() {
        float debit;
        if ((balance > amount) && (balance > 2000)) {
            System.out.println("You can withdraw the amount");
            debit = balance - amount;
            System.out.println("Total balance after the withdrawal: " + debit);
        } else {
            System.out.println("You have insufficient amount in your account");
        }
    }

    public static void main(String[] args) {
        Bank b1 = new Bank();
        b1.getInfo();
        b1.displayInfo();
        b1.credit();
        b1.debit();
    }
}
